azurecompute/src/main/java/org/jclouds/azurecompute/compute/extensions/AzureComputeSecurityGroupExtension.java

```
@@ -0,0 +1,423 @@
+                          return input.configurationSets();
+                       }
+                    })
+                    .transformAndConcat(new Function<Role.ConfigurationSet, Iterable<Role.ConfigurationSet.SubnetName>>() {
+                       @Override
+                       public Iterable<Role.ConfigurationSet.SubnetName> apply(Role.ConfigurationSet input) {
+                          return input.subnetNames();
+                       }
+                    })
+                    .transform(new Function<Role.ConfigurationSet.SubnetName, String>() {
+                       @Override
+                       public String apply(Role.ConfigurationSet.SubnetName input) {
+                          return input.name();
+                       }
+                    })
+                    .toList();
+
+      return FluentIterable.from(subnetNames)
+              .transform(new Function<String, NetworkSecurityGroup>() {
+                 @Override
+                 public NetworkSecurityGroup apply(String input) {
+                    return api.getNetworkSecurityGroupApi().getNetworkSecurityGroupAppliedToSubnet(virtualNetworkName, input);
+                 }
+              })
+              .transform(new NetworkSecurityGroupSecurityGroupFunction())
+              .toSet();
+   }
+
+   @Override
+   public SecurityGroup getSecurityGroupById(String id) {
+      return transformNetworkSecurityGroupToSecurityGroup(id);
+   }
+
+   @Override
+   public SecurityGroup createSecurityGroup(final String name, Location location) {
+      checkNotNull(name, "name");
+      checkNotNull(location, "location");
+
+      final NetworkSecurityGroup networkSecurityGroup = NetworkSecurityGroup.create(name, name, location.getId(), null);
+      String createNSGRequestId =
+              api.getNetworkSecurityGroupApi().create(networkSecurityGroup);
+      operationSucceededPredicate.apply(createNSGRequestId);
```

What if the predicate returns `false`? Should we throw an exception instead of silently continuing? Apply the same reasoning to the rest of operations in this file.